### ISSUE TITLE
regression in pointer event management

### ISSUE THREAD

**Issue by obiot** [melonjs/melonJS#714]
symptom : when running the `shapes` example, it is not possible to move the 3 shapes on the bottom and right side (with tons of shapes), based on where the pointer is when clicking to move it

although i have doubt now, i'm sure at 99% it was working when i added support for the Physic Editor tool (#684), so there must be something wrong somewhere, maybe after #643.

**Comment by parasyte** [comment]
Wow ... e344c2ea9c694f975ec91301d859341d9319846a exposed this bug!

So far I've discovered that `region.containsPoint()` and `bounds.containsPoint` are returning different answers in completely different areas of the screen.

The crazy orange polygon can be moved while the pointer is within its lower-right quadrant, because its `region` is offset by half of its width and height (it looks like a bug with the polygon origin, but I haven't fully investigated.)

I used a conditional breakpoint, as seen below, but you can just as easily simplify the `eventInBounds` check to:

```javascript
var eventInBounds = region.containsPoint(e.gameX, e.gameY);
```

and you get the same poor behavior. The conditional breakpoint is more interesting because it triggers where the actual regions "are located", regardless of what the "shapes example" thinks should be draggable (it also uses the `bounds`).

![screen shot 2015-07-28 at 12 32 58 pm](https://cloud.githubusercontent.com/assets/456942/8941833/e5cc4962-3526-11e5-8770-12ddc300a764.png)

For the two shapes on the right, the `region` is located completely outside of the `bounds`, so it is impossible for the events to trigger on these shapes.

**Comment by parasyte** [comment]
This could be telling: https://github.com/melonjs/melonJS/blob/f8b267a13b9473cc53c1d4bb14cd008db308809e/examples/shapes/data/json/physic-body-editor/test.json#L161 -- The shape definition for the crazy orange polygon puts its origin roughly in the center, which is where `region` is located, in relation to `bounds`.

And well, well, what do you know? Cancel the origin offset stuff here: https://github.com/melonjs/melonJS/blob/d2029510cf4740ed985715948ba5dc227ca6a1df/src/physics/body.js#L248

```javascript
var origin = new me.Vector2d();
```

And pointer events magically work correctly!

So that's the bug... The origin is applied inappropriately for Physics-Body-Editor shapes. It should be applied to the entity's initial position... Not to the polygon vertices.

**Comment by parasyte** [comment]
The other shapes from "CodeAndWeb Physics Editor" have `region` with geometry `width = 200, height = 200`, so the pointer tests are really weird for these, too... the region extends well beyond the boundary.

**Comment by obiot** [comment]
You never cease to amaze me in terms of debugging/bug analysis skills :)

I was on the right track too with the containsPoint function, but still far to understand that the issue was with the origin point.

Two nasty bugs/regression squashed, event and physic editor ! Awesome :)

**Comment by parasyte** [comment]
@obiot What about the `region` geometry not matching the `bounds` geometry with the "CodeAndWeb Physics Editor" shapes? Should we make a followup ticket?

**Comment by obiot** [comment]
oh, did I close this one too quicly ? maybe yes then a follow-up ticket, as this is kind of a different "issue"
